std/algorithm/searching.d

```diff
@@ -233,14 +233,25 @@ bool balancedParens(Range, E)(Range r, E lPar, E rPar,
 if (isInputRange!(Range) && is(typeof(r.front == lPar)))
 {
     size_t count;
-    for (; !r.empty; r.popFront())
+
+    static if (is(Unqual!(ElementEncodingType!Range) == Unqual!E) && isNarrowString!Range)
+    {
+        import std.utf : byCodeUnit;
+        auto rn = r.byCodeUnit;
```

Comment:
I think this requires a test for a `lPar` character - maybe someone used one of the many variations: http://xahlee.info/comp/unicode_matching_brackets.html